stl/inc/spanstream

```diff
@@ -0,0 +1,396 @@
+            if (_Mode & ios_base::ate) {
+                _Mysb::setp(_Buf.data(), _Buf.data() + _Buf.size(), _Buf.data() + _Buf.size());
+            } else {
+                _Mysb::setp(_Buf.data(), _Buf.data(), _Buf.data() + _Buf.size());
+            }
+        }
+
+        if (_Mode & ios_base::in) {
+            _Mysb::setg(_Buf.data(), _Buf.data(), _Buf.data() + _Buf.size());
+        }
+    }
+
+protected:
+    // [spanbuf.virtuals], overridden virtual functions
+    pos_type __CLR_OR_THIS_CALL seekoff(
+        off_type _Off, ios_base::seekdir _Way, ios_base::openmode _Which = ios_base::in | ios_base::out) override {
+        const bool _Sequence_in  = _Which & ios_base::in;
+        const bool _Sequence_out = _Which & ios_base::out;
+        switch (_Way) {
+        case ios_base::beg:
+            {
+                // [spanbuf.virtuals]/4.1 baseoff = 0
+                if (static_cast<size_t>(_Off) > _Buf.size()) { // negative wraparound to positive to save a compare
+                    return pos_type(off_type(-1)); // [spanbuf.virtuals]/5 report failure
+                }
+
+                break;
+            }
+        case ios_base::end:
+            {
+                // [spanbuf.virtuals]/4.3 baseoff =
+                const auto _Baseoff = _Mode & ios_base::out && !(_Mode & ios_base::in)
+                                        ? static_cast<off_type>(_Mysb::pptr() - _Mysb::pbase())
+                                        : static_cast<off_type>(_Buf.size());
+
+                _Off += _Baseoff;
+                if (static_cast<size_t>(_Off) > _Buf.size()) { // negative wraparound to positive to save a compare
+                    return pos_type(off_type(-1)); // [spanbuf.virtuals]/5 report failure
+                }
+
+                break;
+            }
+        case ios_base::cur:
+            {
+                if (_Sequence_in && _Sequence_out) {
+                    return pos_type(off_type(-1)); // report failure
+                } else if (_Sequence_in || _Sequence_out) {
+                    const off_type _Oldoff  = _Sequence_in ? static_cast<off_type>(_Mysb::gptr() - _Mysb::eback())
+                                                           : static_cast<off_type>(_Mysb::pptr() - _Mysb::pbase());
+                    const off_type _Oldleft = static_cast<off_type>(_Buf.size() - _Oldoff);
+                    if (_Off < -_Oldoff || _Off > _Oldleft) { // out of bounds
+                        return pos_type(off_type(-1));
+                    }
+
+                    _Off += _Oldoff;
+                } else {
+                    return pos_type(off_type(-1)); // report failure
+                }
+                break;
+            }
+        default:
+            return pos_type(off_type(-1)); // report failure
+        }
+
+        // [spanbuf.virtuals]/4: For a sequence to be positioned, if its next pointer is a null pointer and newoff
+        // is not equal 0, the positioning operation fails.

```

Comment:
Should say "not equal to" (I know this isn't a literal quote but the grammar should be correct).